From: Free Ekanayaka <freee@debian.org>
Date: Mon, 31 Oct 2016 10:31:46 +0000
Subject: broken-flushWarnings

Fix spurious unit tests failures due to broken behavior in
SynchronousTestCase.flushWarnings, which ends up skipping
warnings. See this code around 1160:

aModule = sys.modules[aFunction.__module__]
filename = inspect.getabsfile(aModule)

if filename != os.path.normcase(aWarning.filename):
    continue
---
 src/incremental/tests/test_version.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/incremental/tests/test_version.py b/src/incremental/tests/test_version.py
index d77396c..abf5af2 100644
--- a/src/incremental/tests/test_version.py
+++ b/src/incremental/tests/test_version.py
@@ -8,6 +8,7 @@ Tests for L{incremental}.
 from __future__ import division, absolute_import
 
 import operator
+import unittest
 
 from incremental import getVersionString, IncomparableVersions
 from incremental import Version, _inf
@@ -101,7 +102,7 @@ class VersionsTests(TestCase):
         Passing 'prerelease' to Version is deprecated.
         """
         Version("whatever", 1, 0, 0, prerelease=1)
-        warnings = self.flushWarnings([self.test_prereleaseDeprecated])
+        warnings = self.flushWarnings()
         self.assertEqual(len(warnings), 1)
         self.assertEqual(
             warnings[0]['message'],
@@ -114,8 +115,7 @@ class VersionsTests(TestCase):
         """
         va = Version("whatever", 1, 0, 0, release_candidate=1)
         va.prerelease
-        warnings = self.flushWarnings(
-            [self.test_prereleaseAttributeDeprecated])
+        warnings = self.flushWarnings()
         self.assertEqual(len(warnings), 1)
         self.assertEqual(
             warnings[0]['message'],
